src/odemis/acq/fastem.py

```diff
@@ -515,12 +511,17 @@ def pre_calibrate(self):
         finally:
             # Blank the beam after the acquisition is done.
             self._scanner.blanker.value = True
-            # set back the initial values on the hardware
+            if asm_config is None:
+                logging.warning("Failed to retrieve asm configuration, configure_asm cannot be executed.")
+            else:
+                configure_hw.configure_asm(self._multibeam, self._descanner, self._detector,
+                                           self._detector.dataflow, asm_config)
+            # put system back into state ready for next task and set calibrated settings
             configure_hw.configure_asm(self._multibeam,
                                        self._descanner,
                                        self._detector,
                                        self._detector.data,
-                                       asm_config_orig,
+                                       asm_config,
```

Comment:
Why do you do the same thing twice in a row?
Also `self._detector.dataflow` probably doesn't exist, it's `self._detector.data`.

Reply:
This indeed is incorrect.